### PR TITLE
Fixed Misra pattern for bsgcontrol

### PR DESCRIPTION
**Pull request by pdsfilho**

### CODE REVIEW

**Review by akafael — APPROVED** [review]
Well done! The code improved a lot after the proposed changes!

File: code/bsgcontrol.c [hunk 1]
```diff
@@ -3,23 +3,36 @@
...
+            stateBSG = BSG_GENERATOR;
+         }
+         //BSG_MOTOR - Torque Assistance
+         else if(getDriverIntention(vehicle) == DRIVE_INTENTION_INCREASE_SPEED){
+            stateBSG = BSG_MOTOR;
+         }
+         if(getDriverIntention(vehicle) == DRIVE_INTENTION_NOTHING) {
+            //BSG_IDLE - Energy Saving
+            stateBSG = BSG_IDLE;
+         }
```
Comment (akafael) [inline code/bsgcontrol.c:31]: It is missing an `else`


Suggestion: add one more block:
```cpp
         }
         else if(getDriverIntention(vehicle) == DRIVE_INTENTION_NOTHING) {
            //BSG_IDLE - Energy Saving
            stateBSG = BSG_IDLE;
         }
         else
         {
            stateBSG = BSG_IDLE;
         }
```
or just combine both cases in a single ELSE statment.

```suggestion
         }
         else
         {
            stateBSG = BSG_IDLE;
         }
```

**Comment by pdsfilho** [comment]
> Well done! The code improved a lot after the proposed changes!

Thanks!

File: code/bsgcontrol.c (hunk 1 above)
Comment (pdsfilho) [reply code/bsgcontrol.c]: done!

**Review by pjmbs — APPROVED** [review]

**Pull request merged**